assets/js/modules/analytics-4/components/common/WebDataStreamSelect.js

```diff
@@ -66,7 +66,7 @@ export default function WebDataStreamSelect( props ) {
 	);
 
 	const webDataStreams = useSelect( ( select ) =>
-		isValidPropertyID( propertyID )
+		isValidPropertyID( propertyID ) && hasModuleAccess !== false
```

Comment:
Should we maybe flip the comparison to be `&& hasModuleAccess === true` (which could be simplified as a simple truthy check) ? Otherwise, the way it is now would be logically true if `hasModuleAccess` was `undefined` (still resolving)  which I don't think we'd want here, correct?